static/ankersrv.js

```diff
@@ -5,20 +5,37 @@ $(function () {
     $("#copyYear").text(new Date().getFullYear());
 
     /**
-     * Redirect page when modal dialog is shown
+     * Handle modal being shown
      */
     var popupModal = document.getElementById("popupModal");
 
     popupModal.addEventListener("shown.bs.modal", function (e) {
-        window.location.href = $("#reload").data("href");
+        const trigger = e.relatedTarget;
+        const modalInner = document.getElementById("modal-inner");
+        modalInner.innerText = trigger.dataset.msg;
+        if (trigger.dataset.href) {
+            window.location.href = trigger.dataset.href;
+        }
+    });
+
+    /**
+     * Opens modal if gcode upload file is present
+     */
+    document.getElementById('gcode-upload').addEventListener('click', function(event) {
```

Comment:
We can use `cash.js` here, to make it easier:

```js
$("#gcode-upload").on("click", function (event) {
...
```

The reason I didn't use it for `popupModal.addEventListener("shown.bs.modal", ...`, is because it doesn't work. Something about `cash.js` and the bootstrap events don't play well together.

Reply:
Sure can.

Reply:
✅